naga/src/back/msl/writer.rs

```diff
@@ -2927,6 +2927,13 @@ impl<W: Write> Writer<W> {
         for statement in statements {
             log::trace!("statement[{}] {:?}", level.0, statement);
             match *statement {
+                crate::Statement::Phony(expr) => {
+                    write!(self.out, "{level}")?;
+                    let name = format!("_phony_{}", expr.index());
+                    self.start_baking_expression(expr, &context.expression, &name)?;
+                    self.put_expression(expr, &context.expression, true)?;
```

Comment:
question: I actually don't know when we should start baking expressions; this is something I need to learn about Naga internals, too.

@gfx-rs/naga, should we be baking expressions here? I presume so, since we may have an arbitrary expression on the RHS of a phony assignment.